app/screens/post_options/bindings/bindings.tsx

```diff
@@ -34,13 +35,31 @@ type Props = {
     };
 }
 
+const fetchBindings = (userId: string, channelId: string, teamId: string, setState: React.Dispatch<React.SetStateAction<AppBinding[] | null>>) => {
+    Client4.getAppsBindings(userId, channelId, teamId).then(
+        (allBindings) => {
+            const headerBindings = allBindings.filter((b) => b.location === AppBindingLocations.POST_MENU_ITEM);
+            const postMenuBindings = headerBindings.reduce((accum: AppBinding[], current: AppBinding) => accum.concat(current.bindings || []), []);
+            setState(postMenuBindings);
+        },
+        () => {/* Do nothing */},
+    ).catch(() => {/* Do nothing */});
+};
+
 const Bindings = injectIntl((props: Props) => {
+    const [bindings, setBindings] = useState(props.bindings);
+
+    if (!bindings) {
+        setBindings([]);
+        fetchBindings(props.currentUser.id, props.post.channel_id, props.teamID, setBindings);
+    }
```

Comment:
This should be in a `useEffect` block since we're changing state here:

```suggestion
    useEffect(() => {
        if (bindings) {
            return;
        }
        
        setBindings([]);
        fetchBindings(props.currentUser.id, props.post.channel_id, props.teamID, setBindings);
    }, bindings);
```